pay-api/tests/unit/api/test_eft_short_name_history.py

```diff
@@ -0,0 +1,151 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests to assure the EFT short names history end-point.
+
+Test-Suite to ensure that the /eft-shortnames/{id}/history endpoint is working as expected.
+"""
+from datetime import datetime
+
+import pytest
+from freezegun import freeze_time
+
+from pay_api.services.eft_short_name_historical import EFTShortnameHistorical as EFTHistoryService
+from pay_api.services.eft_short_name_historical import EFTShortnameHistory as EFTHistory
+from pay_api.utils.enums import EFTHistoricalTypes, PaymentMethod, Role
+from tests.utilities.base_test import factory_eft_shortname, factory_payment_account, get_claims, token_header
+
+
+def setup_test_data(exclude_history: bool = False):
+    """Set up eft short name historical data."""
+    payment_account = factory_payment_account(payment_method_code=PaymentMethod.EFT.value,
+                                              auth_account_id='1234', name='ABC-BRANCH', branch_name='BRANCH').save()
+    short_name = factory_eft_shortname(short_name='TESTSHORTNAME1').save()
+
+    if not exclude_history:
+        EFTHistoryService.create_funds_received(EFTHistory(short_name_id=short_name.id,
+                                                           amount=351.50,
+                                                           credit_balance=351.50)).save()
+
+        EFTHistoryService.create_statement_paid(EFTHistory(short_name_id=short_name.id,
+                                                           amount=351.50,
+                                                           credit_balance=0,
+                                                           payment_account_id=payment_account.id,
+                                                           related_group_link_id=1,
+                                                           statement_number=1234)).save()
+
+        EFTHistoryService.create_statement_reverse(EFTHistory(short_name_id=short_name.id,
+                                                              amount=351.50,
+                                                              credit_balance=351.50,
+                                                              payment_account_id=payment_account.id,
+                                                              related_group_link_id=2,
+                                                              statement_number=1234)).save()
+
+    return payment_account, short_name
+
+
+@pytest.mark.parametrize('result_index, expected_values', [
```

Comment:
awesome thank you, this will save me alot of time commenting out trying to figure out how to fix the tests in the future etc